crates/starknet-devnet/tests/test_minting.rs

```diff
@@ -229,4 +230,33 @@ mod minting_tests {
             .await
             .unwrap();
     }
+
+    #[tokio::test]
+    async fn minting_in_multiple_steps_and_getting_balance_at_each_block() {
+        let devnet =
+            BackgroundDevnet::spawn_with_additional_args(&["--state-archive-capacity", "full"])
+                .await
+                .unwrap();
+
+        // create a block if there are no blocks before starting minting
+        devnet.create_block().await.unwrap();
+
+        let address = Felt::ONE;
+
+        let mint_amount = 1e18 as u128;
+        let unit = FeeUnit::WEI;
+
+        for _ in 0..3 {
+            let BlockHashAndNumber { block_hash, .. } =
+                devnet.json_rpc_client.block_hash_and_number().await.unwrap();
+
+            devnet.mint(address, mint_amount).await;
+
+            let balance_at_block =
+                devnet.get_balance_at_block(&address, BlockId::Hash(block_hash)).await.unwrap();
```

Comment:
Other than `starknet_getStorageAt`, perhaps also add old-block fee estimation, and old-block `starknet_call`, as described on Discord. Of course, if some of the existing tests don't already test this. In an ideal case, we would also test simulation at old block in a similar manner (basically all untested methods which are mentioned in #108).

Reply:
[43a7bea](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/615/commits/43a7beadef46c2a7edde7774fe0d487b20436350)

Reply:
Good additions to our tests. I browsed the existing tests once more and realized we already have a test which tests `getClass` at old blocks (in `test_get_class.rs`: `test_getting_class_at_various_blocks`). Should we leave both tests?

What we still don't have is a simple `call` test. Perhaps one of the existing tests you added could be modified to just make a call on ERC20 and get the balance.

Reply:
get_balance_at_block internally makes a starknet_call

Reply:
Right, didn't see it.

Reply:
[2fbae41](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/615/commits/2fbae415e3117b142d4a6c156e90679fda992813)